main.nf

```diff
@@ -867,42 +1044,480 @@ else {
+
+        output:
+        set val(name), file("${name}.vcf.gz*") into ch_vcf_biscuit_qc ,ch_vcf_for_bedgraph,ch_vcf_for_epiread
+
+        script:
+        filter_duplication = params.skip_deduplication || params.rrbs ? '-u' : ''
+        """
+        biscuit pileup  -q ${task.cpus} $filter_duplication $fasta ${bam} -o ${name}.vcf
+        bgzip -@ ${task.cpus} -f ${name}.vcf
+        tabix -f -p vcf ${name}.vcf.gz
+        """
+    }
+
+    /*
+     * STEP 7 - Create bedgraph file from vcf
+     */
+    process create_Bedgraph {
+        tag "$name"
+        publishDir "${params.outdir}/methylation_extract", mode: params.publish_dir_mode
+
+        input:
+        set val(name), file(vcf) from ch_vcf_for_bedgraph
+
+        output:
+        set val(name), file("*bedgraph" ) into ch_bedgraph_for_intersect_soloWCGW
+
+        script:
+        min_depth = params.min_coverage > 1 ? "${params.min_coverage}" : '1'
+        all_contexts = params.comprehensive ? 'c, cg, ch, hcg, gch' : 'cg'
+        """
+        biscuit vcf2bed -k $min_depth -t $all_contexts  "${vcf[0]}" > "${name}.bedgraph"
+        """
+    }
+
+    if (params.epiread) {
+        if (params.common_dbsnp) {
+            /*
+            * STEP 7.1 - Reformat SNP table for SNP file generation
+            */
+            process reformat_SNP {
+
+                input:
+                file commonSNP_file from ch_commonSNP_for_SNP.collect()
+
+                output:
+                file("reformattedSNP.snv.txt.gz*" ) into ch_reformattedSNP_for_SNP
+
+                script:
+                """
+                less $commonSNP_file | $projectDir/bin/processUcscDbsnp.pl | grep snv | bgzip > reformattedSNP.snv.txt.gz
+                tabix -s 1 -b 2 -e 3 reformattedSNP.snv.txt.gz
+                """
+            }
+        }
+        else {
+            ch_reformattedSNP_for_SNP = Channel.empty()
+        }
+
+
+        /*
+        * STEP 7.2 - Create whitelist for SNP calling
+        */
+        if ( !params.whitelist) {
+            process create_whitelist {
+                tag "$blacklist"
+                publishDir path: "${params.outdir}/reference_genome", saveAs: { params.save_reference ? it : null }, mode: params.publish_dir_mode
+
+                input:
+                file blacklist from ch_blacklist_for_create_whitelist
+                file fasta_index from ch_fasta_for_create_whitelist
+
+                output:
+                file("whitelist.${name}.bed.gz" ) into ch_whitelist_for_SNP, ch_whitelist_for_epiread
+                file "sizes.${name}"
+                script:
+                name = assembly_name - '.fa'
+
+                """
+                cut -f1,2 $fasta_index > sizes.${name}
+                bedtools sort -g sizes.${name} -i $blacklist > ${blacklist.baseName}.sorted.bed
+                bedtools complement -i ${blacklist.baseName}.sorted.bed -g sizes.${name} | grep -v _ | bgzip > whitelist.${name}.bed.gz
```

Comment:
See other comment, I _think_ that this should work.
```suggestion
                """
                cut -f1,2 $fasta_index > sizes.${name}
                bedtools sort -g sizes.${name} -i $blacklist > ${blacklist.baseName}.sorted.bed
                bedtools complement -i ${blacklist.baseName}.sorted.bed -g sizes.${fasta_index.simpleName} | grep -v _ | bgzip > whitelist.${name}.bed.gz
```

https://www.nextflow.io/docs/latest/script.html#check-file-attributes

Reply:
Yes, I changed it with getSimpleName, Thank you!